src/commands/init.ts

```diff
@@ -52,183 +53,138 @@ export default class Init extends Command {
-    const scriptsToBeAdded = addFrontendScripts ? { ...backendScritps, ...packageJsonData.frontendScripts } : backendScritps;
-
-    const conflicts = scripts ? Object.keys(scriptsToBeAdded).filter(name => Object.keys(scripts).includes(name)) : [];
-
-    if (
-      conflicts.length &&
-      !(await cli.confirm(`Script(s) with the name(s) "${conflicts.join('", "')}" already exist(s). Should they be overwritten?`))
-    ) {
-      this.error('Script exits as npm scripts could not be written.', {
-        exit: 11
-      });
+    if (fs.existsSync(cliPath)) {
+      await execa(cliPath, ['new', '.', '--skip-install', '--package-manager', 'npm'], options);
+    } else {
+      await execa('npx', ['@nestjs/cli', 'new', '.', '--skip-install', '--package-manager', 'npm'], options);
     }
 
-    const adjustedPackageJson = {
-      ...this.packageJson(flags),
-      scripts: { ...scripts, ...scriptsToBeAdded },
-      dependencies: { ...dependencies, ...(await this.addDependencies(packageJsonData.dependencies)) },
-      devDependencies: { ...devDependencies, ...(await this.addDependencies(packageJsonData.devDependencies)) }
-    };
+    const pathToMainTs = path.resolve(projectDir, 'src', 'main.ts');
+    const mainTs = fs.readFileSync(pathToMainTs, { encoding: 'utf8' });
+    const modifiedMainTs = mainTs.replace('.listen(3000)', '.listen(process.env.PORT || 3000)');
 
-    fs.writeFileSync(path.resolve(flags.projectDir, 'package.json'), JSON.stringify(adjustedPackageJson, null, 2));
-  }
+    if (mainTs === modifiedMainTs) {
+      cli.warn('Could not adjust listening port to `process.env.PORT`. Please adjust manually.');
+    }
 
-  private async addDependencies(dependencies: string[]): Promise<{ [key: string]: string }> {
-    const versions = await Promise.all(dependencies.map(dependency => this.getVersionOfDependency(dependency)));
-    return dependencies.reduce((result, dependency, index) => ({ ...result, [dependency]: versions[index] }), {} as any);
+    fs.writeFileSync(pathToMainTs, modifiedMainTs);
   }
 
-  private async getVersionOfDependency(dependency: string): Promise<string> {
+  private async getOptions(projectDir: string, startCommand?: string, projectName?: string) {
     try {
-      const defaultOptions = ['view', dependency, 'version'];
-      const version = dependency.includes('@sap')
-        ? execa('npm', [...defaultOptions, '--registry', 'https://npm.sap.com'])
-        : execa('npm', defaultOptions);
-
-      return `^${(await version).stdout}`;
-    } catch (err) {
-      this.warn(`Error in finding version for dependency ${dependency} - use LATEST as fallback.`);
-      return 'latest';
+      const options: { [key: string]: string } = {
+        projectName:
+          projectName ||
+          (await cli.prompt('Enter project name (for use in manifest.yml)', {
+            default: packageJson(projectDir).name
```

Comment:
I would name this function `parsePackageJson`, because then there is no doubt what it does.
Also, you might want to consider calling it once in the beginning to  not read and parse the file twice.